src/lib/text-field/text-field-adapter.ts

```diff
@@ -1,130 +1,187 @@
-import { addClass, listenOwnProperty, getActiveElement, createElementAttributeObserver } from '@tylertech/forge-core';
+import { getShadowElement, listenOwnProperty, randomChars, toggleAttribute } from '@tylertech/forge-core';
+import { FIELD_CONSTANTS, IFieldComponent } from '../field-next';
+import { BaseFieldAdapter, IBaseFieldAdapter } from '../field-next/base/base-field-adapter';
+import { LabelComponent } from '../label';
 import { ITextFieldComponent } from './text-field';
-import { FieldAdapter, IFieldAdapter } from '../field/field-adapter';
-import { TEXT_FIELD_CONSTANTS } from './text-field-constants';
-import { FIELD_CONSTANTS } from '../field/field-constants';
-
-export interface ITextFieldAdapter extends IFieldAdapter {
-  detectTextarea(): void;
+import { TextFieldInputAttributeObserver, TextFieldValueChangeListener, TEXT_FIELD_CONSTANTS } from './text-field-constants';
+
+export interface ITextFieldAdapter extends IBaseFieldAdapter {
+  addRootListener(name: keyof HTMLElementEventMap, listener: EventListener): void;
+  disableInput(disabled: boolean): void;
+  handleDefaultSlotChange(slot: HTMLSlotElement, listener: TextFieldInputAttributeObserver): void;
+  tryAddValueChangeListener(context: unknown, listener: TextFieldValueChangeListener): void;
+  removeValueChangeListener(): void;
+  tryFloatLabel(force?: boolean): void;
+  tryConnectSlottedLabel(slot: HTMLSlotElement): void;
+  connectClearButton(listener: EventListener): void;
+  disconnectClearButton(listener: EventListener): void;
+  toggleClearButtonVisibility(visible: boolean): void;
+  clearInput(): void;
 }
 
-export class TextFieldAdapter extends FieldAdapter implements ITextFieldAdapter {
-  protected _inputElements: Array<HTMLInputElement | HTMLTextAreaElement> = [];
-  protected _inputMutationObserverInstances: MutationObserver[] = [];
+export class TextFieldAdapter extends BaseFieldAdapter implements ITextFieldAdapter {
+  protected readonly _fieldElement: IFieldComponent;
+  private readonly _clearButtonSlotElement: HTMLSlotElement;
+  private _inputElements: (HTMLInputElement | HTMLTextAreaElement)[] = [];
+  private _inputMutationObserver?: MutationObserver;
+  private _destroyValueChangerListeners: (() => void)[] = [];
```

Comment:
Are we cleaning these up when the element is disconnected?

Reply:
Yeah, we should do that